### PR TITLE
22580 - return effective date and NR expiry date in reviews response

### PR DESCRIPTION
**Pull request by ketaki-deodhar**
*Issue #:* /bcgov/entity#22580

*Description of changes:*

- return effective date and NR expiry date in reviews response
- update call for getting token (used from `AccountService`)
- sample json 
[Sample json.txt](https://github.com/user-attachments/files/16700827/Sample.json.txt)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

File: legal-api/src/legal_api/services/namex.py [hunk 1]
```diff
@@ -73,6 +65,24 @@ def query_nr_number(identifier: str):
...
+        # Perform proxy call with identifiers (e.g. NR 1234567, NR 1234568)
+        nr_response = requests.post(namex_url + 'requests/search',
+                                    json={'identifiers': identifiers},
+                                    headers={
+                                        'Content-Type': 'application/json',
+                                        'Authorization': 'Bearer ' + token
+                                    })
+
+        return nr_response
+
```
Comment (ketaki-deodhar) [inline legal-api/src/legal_api/services/namex.py:85]: call to namex to retrieve multiple NRs data

**Comment by ketaki-deodhar** [comment]
@vysakh-menon-aot I am fixing the unit test in the background and will push changes soon

File: legal-api/src/legal_api/models/review.py [hunk 2]
```diff
@@ -79,18 +82,27 @@ def get_review(cls, filing_id) -> Review:
...
-        ]
+        for review, effective_date in results:
+            future_effective_date = ''
+            if effective_date > datetime.now(timezone.utc):
+                future_effective_date = LegislationDatetime.format_as_legislation_date(effective_date)
+
+            result.append({
+                'review': review.json,
+                'effectiveDate': future_effective_date
+            })
```
Comment (vysakh-menon-aot) [inline legal-api/src/legal_api/models/review.py:102]:
```suggestion
            result.append({
                **review.json,
                'effectiveDate': future_effective_date
            })
```

File: legal-api/src/legal_api/services/namex.py [hunk 3]
```diff
@@ -50,20 +52,10 @@ class State(Enum):
     @staticmethod
     def query_nr_number(identifier: str):
         """Return a JSON object with name request information."""
-        auth_url = current_app.config.get('NAMEX_AUTH_SVC_URL')
-        username = current_app.config.get('NAMEX_SERVICE_CLIENT_USERNAME')
-        secret = current_app.config.get('NAMEX_SERVICE_CLIENT_SECRET')
         namex_url = current_app.config.get('NAMEX_SVC_URL')
 
         # Get access token for namex-api in a different keycloak realm
```
Comment (vysakh-menon-aot) [inline legal-api/src/legal_api/services/namex.py:57]: can remove this

File: legal-api/src/legal_api/services/namex.py [hunk 4]
```diff
@@ -73,6 +65,24 @@ def query_nr_number(identifier: str):
 
         return nr_response
 
+    @staticmethod
+    def query_nr_numbers(identifiers):
+        """Return a JSON object with name request information."""
+        namex_url = current_app.config.get('NAMEX_SVC_URL')
+
+        # Get access token for namex-api in a different keycloak realm
```
Comment (vysakh-menon-aot) [inline legal-api/src/legal_api/services/namex.py:73]: can remove this

File: legal-api/src/legal_api/resources/v2/admin/reviews.py [hunk 5]
```diff
@@ -32,9 +32,49 @@ def get_reviews():
...
+
+    return jsonify(result), HTTPStatus.OK
+
+
+def get_applicable_nr_numbers(reviews):
+    """Return list of NR numbers of review with status CHANGE_REQUESTED/AWAITING_REVIEW/RESUBMITTED."""
+    nr_numbers = []
+    for review in reviews:
+        current_nr = review['review']['nrNumber']
+        current_status = review['review']['status']
```
Comment (vysakh-menon-aot) [inline legal-api/src/legal_api/resources/v2/admin/reviews.py:53]:
```suggestion
        current_nr = review['nrNumber']
        current_status = review['status']
```

File: legal-api/src/legal_api/resources/v2/admin/reviews.py [hunk 6]
```diff
@@ -32,9 +32,49 @@ def get_reviews():
...
+    result = Review.get_paginated_reviews(page, limit)
+    reviews = result['reviews']
 
-    return jsonify(reviews), HTTPStatus.OK
+    if not reviews:
+        return jsonify({'message': 'Reviews not found.'}), HTTPStatus.NOT_FOUND
+
+    nr_numbers = get_applicable_nr_numbers(reviews)
+    nr_expiry_dates = get_expiry_date_for_each_nr(nr_numbers)
+    update_reviews(reviews, nr_expiry_dates)
```
Comment (vysakh-menon-aot) [inline legal-api/src/legal_api/resources/v2/admin/reviews.py:43]:
```suggestion
    if nr_numbers:
        nr_expiry_dates = get_expiry_date_for_each_nr(nr_numbers)
        update_reviews(reviews, nr_expiry_dates)
```

**Comment by ketaki-deodhar** [comment]
> @vysakh-menon-aot I am fixing the unit test in the background and will push changes soon

Tests run successfully locally but somehow test check is failing in this PR. Not sure what is triggering this. I tried re-running failed job with no luck. Any pointers?

image

File: legal-api/src/legal_api/resources/v2/admin/reviews.py [hunk 7]
```diff
@@ -32,9 +32,50 @@ def get_reviews():
     """Return a list of reviews."""
     page = int(request.args.get('page', 1))
     limit = int(request.args.get('limit', 10))
-    reviews = Review.get_paginated_reviews(page, limit)
+    result = Review.get_paginated_reviews(page, limit)
+    reviews = result['reviews']
 
-    return jsonify(reviews), HTTPStatus.OK
+    if not reviews:
+        return jsonify({'message': 'Reviews not found.'}), HTTPStatus.NOT_FOUND
```
Comment (vysakh-menon-aot) [inline legal-api/src/legal_api/resources/v2/admin/reviews.py:39]: I don't think this is required

**Comment by vysakh-menon-aot** [comment]
> > @vysakh-menon-aot I am fixing the unit test in the background and will push changes soon
> 
> Tests run successfully locally but somehow test check is failing in this PR. Not sure what is triggering this. I tried re-running failed job with no luck. Any pointers you might suggest?
> 
> image

You need to mock the namex api call (like this https://github.com/bcgov/lear/blob/d540fdf8387824c76224111bf1b916b689c01e4d/legal-api/tests/unit/resources/v2/test_admin/test_reviews.py#L188)

**Review by vysakh-menon-aot — APPROVED** [review]

**Pull request merged**